lib/ansible/errors/__init__.py

```diff
@@ -50,14 +55,42 @@ class AnsibleError(Exception):
     which should be returned by the DataLoader() class.
     '''
 
-    def __init__(self, message="", obj=None, show_content=True, suppress_extended_error=False, orig_exc=None):
+    def __init__(
+        self,
+        message="",
+        obj=None,
+        show_content=True,
+        suppress_extended_error=False,
+        orig_exc=_UNDERLYING_EXC_SENTINEL,
+    ):
         super(AnsibleError, self).__init__(message)
 
         self._show_content = show_content
         self._suppress_extended_error = suppress_extended_error
         self._message = to_native(message)
         self.obj = obj
-        self.orig_exc = orig_exc
+        if orig_exc is not _UNDERLYING_EXC_SENTINEL:
```

Comment:
could just leave the original None and test for that `if orig_exec is not None` , a lot less change just keeping the 'old sentinel'.

Reply:
Yep, that's technically true. However, I like naming data used in code and I wanted to be verbose about its semantics. Moreover, it's important for the case of the “argument not provided” to differ from an explicit `AnsibleError(..., orig_exc=None)`. It is customary to use `None` in cases where the caller wants to intentionally suppress exception context chaining. That's why you can sometimes meet `raise AnException from None` that is spelled out in `except:` blocks.

Reply:
that is not something we currently do, weird to add new functionality in something you with to deprecate/remove